crates/rpc/src/v04/method/trace_transaction.rs

```diff
@@ -0,0 +1,173 @@
+use anyhow::Context;
+use pathfinder_common::{BlockHash, BlockId, TransactionHash};
+use pathfinder_executor::{CallError, Transaction};
+use primitive_types::U256;
+use serde::{Deserialize, Serialize};
+use tokio::task::JoinError;
+
+use crate::{
+    compose_executor_transaction,
+    context::RpcContext,
+    error::{RpcError, TraceError},
+    executor::ExecutionStateError,
+};
+
+use super::simulate_transactions::dto::TransactionTrace;
+
+#[derive(Deserialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct TraceTransactionInput {
+    pub transaction_hash: TransactionHash,
+}
+
+#[derive(Debug, Serialize, Eq, PartialEq)]
+pub struct TraceTransactionOutput(pub TransactionTrace);
+
+#[derive(Debug)]
+pub enum TraceTransactionError {
+    InvalidTxnHash,
+    NoTraceAvailable(TraceError),
+    Internal(anyhow::Error),
+}
+
+impl From<ExecutionStateError> for TraceTransactionError {
+    fn from(value: ExecutionStateError) -> Self {
+        match value {
+            ExecutionStateError::BlockNotFound => Self::NoTraceAvailable(TraceError::Rejected),
+            ExecutionStateError::Internal(e) => Self::Internal(e),
+        }
+    }
+}
+
+impl From<CallError> for TraceTransactionError {
+    fn from(value: CallError) -> Self {
+        match value {
+            CallError::ContractNotFound | CallError::InvalidMessageSelector => {
+                Self::NoTraceAvailable(TraceError::Rejected)
+            }
+            CallError::Reverted(_) => Self::NoTraceAvailable(TraceError::Received),
```

Comment:
I don't think `executor::trace_one` ever returns this -- and its mapping to `NoTraceAvailable` is also questionable.